mllib/src/main/scala/org/apache/spark/ml/feature/RFormula.scala

```diff
@@ -163,12 +163,20 @@ class RFormula @Since("1.5.0") (@Since("1.5.0") override val uid: String)
     }.toMap
 
     // Then we handle one-hot encoding and interactions between terms.
+    var hasReferenceCategory = false
```

Comment:
because then the 2nd condition `!hasReferenceCategory` is always true on L175 https://github.com/apache/spark/pull/12414/files#diff-bedaa993ebc2cc2e0d496859095270feR175